server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -0,0 +1,465 @@
+        Map<String, List<String>> forcedAwarenessAttributes = new HashMap<>();
+        Map<String, Settings> forceGroups = forceSettings.getAsGroups();
+        for (Map.Entry<String, Settings> entry : forceGroups.entrySet()) {
+            List<String> aValues = entry.getValue().getAsList("values");
+            if (aValues.size() > 0) {
+                forcedAwarenessAttributes.put(entry.getKey(), aValues);
+            }
+        }
+        this.forcedAwarenessAttributes = forcedAwarenessAttributes;
+    }
+
+    /**
+     * Starts the new decommission request and registers the metadata with status as {@link DecommissionStatus#INIT}
+     * or the last known status if not {@link DecommissionStatus#FAILED}
+     * Once the status is updated, it tries to exclude to-be-decommissioned cluster manager nodes from Voting Configuration
+     *
+     * @param decommissionAttribute register decommission attribute in the metadata request
+     * @param listener register decommission listener
+     */
+    public void startDecommissionAction(
+        final DecommissionAttribute decommissionAttribute,
+        final ActionListener<ClusterStateUpdateResponse> listener
+    ) {
+        // register the metadata with status as DECOMMISSION_INIT as first step
+        clusterService.submitStateUpdateTask("decommission [" + decommissionAttribute + "]", new ClusterStateUpdateTask(Priority.URGENT) {
+            @Override
+            public ClusterState execute(ClusterState currentState) {
+                // validates if correct awareness attributes and forced awareness attribute set to the cluster before starting action
+                validateAwarenessAttribute(decommissionAttribute, awarenessAttributes, forcedAwarenessAttributes);
+                DecommissionAttributeMetadata decommissionAttributeMetadata = currentState.metadata().decommissionAttributeMetadata();
+                // check that request is eligible to proceed
+                ensureEligibleRequest(decommissionAttributeMetadata, decommissionAttribute);
+                decommissionAttributeMetadata = new DecommissionAttributeMetadata(decommissionAttribute);
+                logger.info("registering decommission metadata [{}] to execute action", decommissionAttributeMetadata.toString());
+                return ClusterState.builder(currentState)
+                    .metadata(Metadata.builder(currentState.metadata()).decommissionAttributeMetadata(decommissionAttributeMetadata))
+                    .build();
+            }
+
+            @Override
+            public void onFailure(String source, Exception e) {
+                logger.error(
+                    () -> new ParameterizedMessage(
+                        "failed to start decommission action for attribute [{}]",
+                        decommissionAttribute.toString()
+                    ),
+                    e
+                );
+                listener.onFailure(e);
+            }
```

Comment:
What will be the response to the user if user started a second decommission while the first one was `IN-PROGRESS`

Reply:
> What will be the response to the user if user started a second decommission while the first one was IN-PROGRESS

User will see `DecommissioningFailedException` with msg = "same request is already in status ["+decommissionAttributeMetadata.status()+"], please wait for it to complete"